pkg/recipes/terraform/module.go

```diff
@@ -53,13 +50,21 @@ type moduleInspectResult struct {
 // localModuleName is the name of the module specified in the configuration used to download the module.
 // It uses terraform-config-inspect to load the module from the directory. An error is returned if the module
 // could not be loaded.
-func inspectModule(workingDir, localModuleName string) (*moduleInspectResult, error) {
+func inspectModule(workingDir, localModuleName string, templatePath string) (*moduleInspectResult, error) {
+	moduleRootDir := filepath.Join(".terraform/modules", localModuleName)
 	result := &moduleInspectResult{ContextVarExists: false, RequiredProviders: []string{}, ResultOutputExists: false, Parameters: map[string]any{}}
 
 	// Modules are downloaded in a subdirectory in the working directory.
 	// Name of the module specified in the configuration is used as subdirectory name.
 	// https://developer.hashicorp.com/terraform/tutorials/modules/module-use#understand-how-modules-work
-	mod, diags := tfconfig.LoadModule(filepath.Join(workingDir, moduleRootDir, localModuleName))
+	//
+	// If the template path is for a submodule, we'll add the submodule path to the module root directory.
+	subModDex := strings.LastIndex(templatePath, "//")
+	// Make sure that we don't mistake the double slashes in a '://' scheme as a submodule indicator
+	if subModDex != -1 && templatePath[subModDex-1:subModDex+len("//")] != "://" {
```

Comment:
I don't have full context but this code may give error if subModDex ==0. Is that a possible case? Do we need to update the condition?

Reply:
Ah, good catch. There shouldn't be any Terraform sources that can start with `//` but this should still handle that case. Will update this check

Reply:
I wonder if this works for non https module source kind.
E.g: github.com/project/module

Reply:
I was thinking as @lakshmimsft said, that would hit `index out of range` at `templatePath[subModDex-1:subModDex+len("//")]`  if it was not a submodule

Reply:
Ended up using an existing TF function so this check is no longer needed